pinBowling/src/game.js

```diff
@@ -0,0 +1,40 @@
+function Game (pin, defaultScore) {
+  'use strict'
+  this.pin = pin || new Pin ();
+  this.frame = this.pin.frame;
+  this._DEFAULT_SCORE = defaultScore || 0;
+  this.score = this._DEFAULT_SCORE;
+  this._loopLimit = 0;
+  this._initialPinsThere = this.pin._initialPinsThere;
+  this._strikeBonusTime = 0;
+  this._bonusCollection = [];
+}
+
+// Need to unravel below function or start from scratch
+
+Game.prototype.pinsHit = function (number) {
+  this.pin.pinsHit(number);
+  this._increaseScore(number);
+  if (this._bonusCollection.length !== 0) {
+  for (var bonus in this._bonusCollection) {bonus.timeLimit--;}
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.